animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -370,3 +384,272 @@ def calculateLinkedVertices(int verticesAmount, EdgeIndicesList edges):
+        else:
+            triAmount += 1
+
+    cdef unsigned int *_oldIndices = polygons.indices.data
+    cdef unsigned int *_oldPolyStarts = polygons.polyStarts.data
+
+    cdef PolygonIndicesList newPolygons = PolygonIndicesList(
+                                          indicesAmount = 3 * triAmount,
+                                          polygonAmount = triAmount)
+    cdef unsigned int *_newIndices = newPolygons.indices.data
+    cdef unsigned int *_newPolyStarts = newPolygons.polyStarts.data
+    cdef unsigned int *_newPolyLengths = newPolygons.polyLengths.data
+
+    cdef Py_ssize_t j, index, triIndex, start
+
+    index = 0
+    triIndex = 0
+    for i in range(polygonAmount):
+        start = _oldPolyStarts[i]
+        for j in range(_oldPolyLengths[i] - 2):
+            _newPolyStarts[triIndex] = index
+            _newPolyLengths[triIndex] = 3
+            triIndex += 1
+
+            _newIndices[index] = _oldIndices[start]
+            _newIndices[index + 1] = _oldIndices[start + 1 + j]
+            _newIndices[index + 2] = _oldIndices[start + 2 + j]
+            index += 3
+
+    return newPolygons
+
+@cython.cdivision(True)
+def triangulatePolygonsUsingEarClipMethod(Vector3DList oldVertices, PolygonIndicesList polygons):
+    cdef unsigned int *_oldPolyLengths = polygons.polyLengths.data
+    cdef Py_ssize_t polygonAmount = polygons.getLength()
+    cdef Py_ssize_t i, triAmount, polyLength
+
+    # Total number of triangle polygons
+    triAmount = 0
+    for i in range(polygonAmount):
+        polyLength = _oldPolyLengths[i]
+        if polyLength > 3:
+            triAmount += polyLength - 2
+        else:
+            triAmount += 1
+
+    cdef unsigned int *_oldIndices = polygons.indices.data
+    cdef unsigned int *_oldPolyStarts = polygons.polyStarts.data
+
+    cdef PolygonIndicesList newPolygons = PolygonIndicesList(
+                                          indicesAmount = 3 * triAmount,
+                                          polygonAmount = triAmount)
+    cdef unsigned int *_newIndices = newPolygons.indices.data
+    cdef unsigned int *_newPolyStarts = newPolygons.polyStarts.data
+    cdef unsigned int *_newPolyLengths = newPolygons.polyLengths.data
+
+    cdef Vector3 preVertex, earVertex, nexVertex, ab, bc, v1, v2, v3
+    cdef LongList neighbors, indices, angles, sortAngles
+    cdef Vector3DList polyVertices
+    cdef float angle
+    cdef Py_ssize_t polyStart, triIndex, polyIndex
+    cdef Py_ssize_t j, k, amount, index, preIndex, earIndex, nexIndex
+
+    triIndex = 0
+    polyIndex = 0
+    for i in range(polygonAmount):
+        polyLength = _oldPolyLengths[i]
+        polyStart = _oldPolyStarts[i]
+        neighbors = LongList(length = polyLength)
+        indices = LongList(length = polyLength)
+        for j in range(polyLength):
+            neighbors.data[j] = _oldIndices[polyStart + j]
+            indices.data[j] = j
+
+        polyVertices = transformVerticesOfPolygon(oldVertices, neighbors)
```

Comment:
Should we call this something more meaningful? Maybe `projectPolygonVertices`.

Reply:
Done.